source/common/upstream/upstream_impl.h

```diff
@@ -224,6 +224,13 @@ class HostImpl : public HostDescriptionImpl,
   void weight(uint32_t new_weight) override;
   bool used() const override { return used_; }
   void used(bool new_used) override { used_ = new_used; }
+  bool warmed() const override {
+    if (cluster_ != nullptr && cluster_->warmHosts()) {
```

Comment:
Why bother checking this vs. just check the flag directly out of curiosity?

Reply:
As written the flag is always set, even if the config flag is not set. I guess the alternative solution is to have the health checker check the flag to determine whether the flag should be set or not and then just read the flag. I think that sounds a bit cleaner (moves the complexity out `Host`), so I'll try to give that a go